f3/granite.go

```diff
@@ -606,23 +673,26 @@ type quorumState struct {
 	// CID of each chain received, by sender. Allows detecting and ignoring duplicates.
 	received map[ActorID]senderSent
 	// The power supporting each chain so far.
-	chainPower map[TipSetID]chainPower
+	chainSupport map[TipSetID]chainSupport
 	// Total power of all distinct senders from which some chain has been received so far.
 	sendersTotalPower uint
 	// Table of senders' power.
 	powerTable PowerTable
+	// justifiedMessages stores the received evidences for each message, indexed by the message's head CID.
+	justifiedMessages map[TipSetID]AggEvidence
 }
 
-// The set of chain heads from one sender, and that sender's power.
+// The set of chain heads from one sender and associated signature, and that sender's power.
 type senderSent struct {
-	heads []TipSetID
-	power uint
+	heads map[TipSetID][]byte
+	power uint //TODO this is unnecessary since quorumState has powerTable. Remove.
 }
 
-// A chain value and the total power supporting it.
-type chainPower struct {
+// A chain value and the total power supporting it
+type chainSupport struct {
 	chain           ECChain
 	power           uint
+	signers         bitfield.BitField
```

Comment:
It'll probably be easier to understand if this just a `map[ActorID]{}` here. Or `map[int]`, one of the problems is I'm not sure what the values represent. `Bitfield` just has that internally anyway. We can build a Bitfield for serialization when necessary.

Reply:
used `map[ActorID]struct{}`